common-lib/src/cmr/common/generics.clj

```diff
@@ -117,3 +119,118 @@
                        (get (json/parse-string index-raw true) :SubConceptType "X")))))
           {}
           (latest-approved-documents)))
+
+(defn read-generic-doc-file
+  "Return the specific schema's documentation files given the schema keyword name and version number.
+   if the file cannot be read, return an empty string which will have no impact on the API document.
+   Parameters:
+   * file-name: [ingest | search]
+   * generic-keyword: [:grid | ...]
+   * generic-version: 0.0.1
+   Returns: string"
+  [file-name generic-keyword generic-version]
+  (try
+    (-> "schemas/%s/v%s/%s.md"
+        (format (name generic-keyword) generic-version (name file-name))
+        (io/resource)
+        (slurp))
+    (catch Exception e (info (format "generic %s was skipped" generic-keyword))
+           (str ""))))
+
+(defn all-generic-docs
+  "Parse over all of the generic documents and return their combined markdown as a string
+  Parameters:
+  * file-name: [ingest | search]
+  Returns: string"
+  [file-name]
+  (string/join (seq (for [[k,v] (latest-approved-documents)]
+                      (read-generic-doc-file file-name k (str v))))))
+
+(def search-depth-map (hash-map 3 0 4 4 5 8))
+(def ingest-depth-map (hash-map 3 8))
+
+(defn get-toc-headers-from-markdown
+  "From the markdown file string of either the ingest or search,
+  dyanamically create the table of content
+  Parameters:
+   * generic-markdown: markdown file for a generic's ingest.md or search.md
+   Returns: string"
+  [generic-markdown]
+  (re-seq #"\#+\s+<a\s.*" generic-markdown))
+
+(defn fill-in-generic-name
+  "Return the html for the table of contents given a generic concept
+   Parameters:
+   * table-of-contents-template: [ingest-table-of-contents-template | search-table-of-contents-template]
+   * generic-type: grid, dataqualitysummary etc
+   Returns: string"
+  [table-of-contents-template generic-type]
+  (-> table-of-contents-template
+      (string/replace #"%generic%" generic-type)
+      (string/replace #"%uc-generic%" (string/capitalize generic-type))
+      (string/replace #"%plural-generic%" (inf/plural generic-type))
+      (string/replace #"%uc-plural-generic%" (inf/plural (string/capitalize generic-type)))))
+
+(defn build-markdown-toc
+  "Get the markdown file for a generic based on the api string being read in
+   Depth is a measure of how many spaces we need from the start of the line for the
+   markdown
+   Parameters:
+   * depth: integer a count of the number of spaces
+   * content: string, the content for the toc
+   * link: string, the link to the location in the document"
+  [depth content link]
+  (str (apply str (repeat depth " ")) (format "* [%s](#%s)\n" link content)))
```

Comment:
I like the repeat and think it should be a function for future use

Reply:
This was removed in favor of a function that is passed by apps using the static.clj to build their API doc